okhttp/src/commonJvmAndroid/kotlin/okhttp3/CompressionInterceptor.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2025 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.ResponseBody.Companion.asResponseBody
+import okhttp3.internal.http.promisesBody
+import okio.BufferedSource
+import okio.GzipSource
+import okio.Source
+import okio.buffer
+
+/**
+ * Transparent Compressed response support.
+ *
+ * The algorithm map will be turned into a heading such as "Accept-Encoding: br;q=1.0, gzip;q=0.8, *;q=0.1"
+ *
+ * If [algorithms] is empty returns "identity" disabling any later compression.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Accept-Encoding
+ */
+open class CompressionInterceptor(
+  val algorithms: Map<DecompressionAlgorithm, Double?>,
```

Comment:
An alternative to this Map, would be for DecompressionAlgorithm to have a priority, defaulting to null. But also a `withPriority()` method to clone and assign.

Reply:
I preferred the varargs. We don’t need a priority if we have a sequence